### PR TITLE
edit annotation label fields remain on return to main page

### PR DESCRIPTION
**Pull request by sebastus**
### What does this PR do?

The motivation for this PR is a bug - if a user edits an annotation label but returns to the main page before committing the edit, the textarea's used for the edit remain on the screen. 

Due to the reliability deficit of "unload" page events in the browser, I elected to rewrite the existing javascript to a more typescript/react mode. 

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: client/src/pages/recording-view/components/annotation/annotation-viewer.tsx [hunk 1]
```diff
@@ -24,6 +77,11 @@ const AnnotationViewer = ({ currentFFT }: AnnotationViewerProps) => {
     setSelectedAnnotation,
   } = useSpectrogramContext();
   const lower_freq = meta.getCenterFrequency() - meta.getSampleRate() / 2;
+  const [editAnnotationLabelY, setEditAnnotationLabelY] = useState(null);
+  const [editAnnotationLabelX, setEditAnnotationLabelX] = useState(null);
+  const [editAnnotationLabelId, setEditAnnotationLabelId] = useState(null);
+  const [editAnnotationLabelText, setEditAnnotationLabelText] = useState(null);
+  const [editAnnotationLabelChanged, setEditAnnotationLabelChanged] = useState(false);
```
Comment (Jade-Codes) [inline client/src/pages/recording-view/components/annotation/annotation-viewer.tsx:85]: would it be easier to do@

const annotationTextarea = useRef(null);
{editAnnotationLabelY && editAnnotationLabelId !== null && (< textarea ref={annotationTextarea} /> )} ?

File: client/src/pages/recording-view/components/annotation/annotation-viewer.tsx [hunk 2]
```diff
@@ -2,11 +2,64 @@
...
+          setEditAnnotationLabelChanged(true);
+        }
+      }
+    }
+  };
+  const top = y + 'px';
+  const left = x + 'px';
+  const msgTop = (y - 30).toString() + 'px';
+  return (
+    <Group>
```
Comment (Jade-Codes) [inline client/src/pages/recording-view/components/annotation/annotation-viewer.tsx:48]: You don't need to do <Group> or <Html>

<>

</>

Will suffice

File: client/src/pages/recording-view/components/annotation/annotation-viewer.tsx [hunk 3]
```diff
@@ -2,11 +2,64 @@
...
+        <textarea
+          placeholder={labelText}
+          style={{ top: top, left: left, width: '200px', fontSize: '16px', position: 'absolute' }}
+          rows={1}
+          className={'text-base-100'}
+          id={id}
+          onKeyDown={onKeyDown}
+        ></textarea>
+      </Html>
+      <Text text="Hit Enter to Finish" fontFamily="serif" fontSize={24} x={x} y={y - 30} fill={'black'}></Text>
```
Comment (Jade-Codes) [inline client/src/pages/recording-view/components/annotation/annotation-viewer.tsx:59]: Do we need to be using Konva elements here? I would have thought a textarea/input would be fine

Comment (sebastus) [reply client/src/pages/recording-view/components/annotation/annotation-viewer.tsx]: I tried really hard to use the second textarea for the "press enter to finish", but couldn't figure out how to get the font in the right color. Then I noticed that I don't really need the textarea for that. A konva Text works just fine.

File: client/src/pages/recording-view/components/annotation/annotation-viewer.tsx (hunk 2 above)
Comment (sebastus) [reply client/src/pages/recording-view/components/annotation/annotation-viewer.tsx:36]: I think I understand what you mean, and changed it. Please confirm.

File: client/src/pages/recording-view/components/annotation/annotation-viewer.tsx (hunk 1 above)
Comment (sebastus) [reply client/src/pages/recording-view/components/annotation/annotation-viewer.tsx:78]: maybe, but I would need more detail to see what you mean.

**Pull request closed without merging**